Assets/Scripts/Frame.cs

```diff
@@ -1,189 +0,0 @@
-        }
-        TotalRolls.Add(roll);
-        UpdateCumulativeFrameScores();
-    }
-
-    private void UpdateCumulativeFrameScores()
-    {
-        FrameScores.Clear();
-        for (int i = 0; i < FrameList.Count; i++)
-        {
-            var frame = FrameList[i];
-            
-            if (frame.GetFrameScore() != null)
-            {
-                if (i == 0)
-                {
-                    FrameScores.Add(frame.GetFrameScore());
-                }
-                else
-                {
-                    FrameScores.Add(frame.GetFrameScore() + FrameScores[i - 1]);
-                }
-            }
-        }
-    }
-
-    public List<int> GetRolls()
-    {
-        return rolls;
-    }
-
-    public bool IsFrameComplete()
-    {
-        //last frame
-        if (lastFrame)
-        {
-            //no extra bowl in last frame
-            if (rolls.Count >= MaxRollsPerFrame && !ExtraBowlAwarded())
-            {
-               
-                return true;
-            }
-            
-            return (rolls.Count >= MaxRollsInFinalFrame);
-        } 
-        
-        //strike ends frame
-        if (rolls[0] == TotalPinsPerFrame)
-        {
-            return true;
-        }
-        
-        return (rolls.Count >= MaxRollsPerFrame);
-    }
-
-    public void AddRoll(int roll)
-    {
-        if(!IsFrameComplete())
-        {
-            rolls.Add(roll);
-            if (IsSpare()) //needed for spare scoring
-            {
-                strikeOrSpareIndex = TotalRolls.Count;
-            }
-            TotalRolls.Add(roll);
-            UpdateCumulativeFrameScores();
-        }
-        else
-        {
-            Debug.LogWarning("Trying to add a roll to a finished frame");
-        }
-    }
-
-    public int? GetFrameScore()
-    {
-        if (!IsFrameComplete())
-        {
-            return null;
-        }
-        
-        var additionalRolls = 0;
-        if (IsStrike())
-        {
-            additionalRolls = StrikeScoreRolls;     
-        }
-        else if (IsSpare())
-        {
-            additionalRolls = SpareScoreRolls;
-        }
-
-        var score = 0;
-        if (additionalRolls > 0)
-        {
-            //check to see if enough frames bowled
-            if (TotalRolls.Count <= strikeOrSpareIndex + additionalRolls)
-            {
-                return null;
-            }
-            
-            score = 10;
-            for (int i = 1; i <= additionalRolls; i++)
-            {
-                score += TotalRolls[strikeOrSpareIndex + i];
-            }
-            return score;
-        }
-        
-        return rolls.Sum(); 
-    }
-
-    public bool IsStrike()
-    {
-        return rolls[0] == 10;
-    }
-
-    public bool IsSpare()
-    {
-        return (rolls[0] + rolls[1] == 10);
-    }
-
-    private bool ExtraBowlAwarded()
-    {
-        if (lastFrame)
-        {
-            //strike or spare
-            if (rolls[0] == 10)
```

Comment:
Make sure to use your utility methods for IsStrike or IsSpare otherwise you could introduce bugs when changing them, and not remembering there were other examples.